admin/spec/back-end/controllers/results.spec.js

```diff
@@ -138,10 +142,41 @@ describe('results controller:', () => {
       expect(resultService.getSchoolScore).toHaveBeenCalled()
       expect(groupService.getGroups).toHaveBeenCalled()
       expect(headteacherDeclarationService.isHdfSubmittedForCurrentCheck).toHaveBeenCalled()
-      expect(schoolHomeFeatureEligibilityPresenter.isResultsPageAccessible).toHaveBeenCalled()
+      expect(schoolHomeFeatureEligibilityPresenter.isResultsPageAccessibleForSubmittedHdfs).toHaveBeenCalled()
       expect(resultPresenter.getResultsViewData).not.toHaveBeenCalled()
       expect(res.render).toHaveBeenCalledWith('results/view-unavailable-results', { breadcrumbs: undefined })
     })
+    it('renders results view page when hdf record for school is not found but datetime for unsubmitted hdfs has passed', async () => {
+      const res = getRes()
+      const req = getReq(reqParams)
+      spyOn(res, 'render')
+      spyOn(checkWindowV2Service, 'getActiveCheckWindow').and.returnValue({ id: 1 })
```

Comment:
The amount of setup here - the number of spy calls - is a bit high.  It feels like the controller is doing too much work.

Reply:
@jon-shipley what are you proposing for this?
This has been merged work into results controller for a while now, perhaps we could extract some work to simplify this taken into account that some of this calls will be removed once redis cache kicks in

Reply:
@GeorgeChatzigiannis - I have only read the diff, but this is clearly a complicated controller.  I wonder if it could be simplified?

Reply:
Not by much in my opinion.
These are the data feeds that are required for the results page:
- checkWindow
- pupils
- groups
- school score
- hdf
- results list page availability

Imagine each one has a dedicated service call.
Feel free to provide suggestions.